conference.js

```diff
@@ -971,7 +971,18 @@ export default {
                     // Rollback the audio muted status by using null track
                     return null;
                 })
-                .then(audioTrack => this.useAudioStream(audioTrack));
+                .then(async audioTrack => {
+                    // If screen audio is already being sent, mix it with the newly created audio track.
+                    if (this._desktopAudioStream) {
+                        // Detach the _desktopAudioStream from the room before setting it as an effect.
+                        await room.replaceTrack(this._desktopAudioStream, null);
+                        this._mixerEffect = new AudioMixerEffect(this._desktopAudioStream);
+                        logger.debug('Mixing new audio track with existing screen audio track.');
+                        await audioTrack.setEffect(this._mixerEffect);
+                    }
+
+                    this.useAudioStream(audioTrack);
+                });
         } else {
             muteLocalAudio(mute);
```

Comment:
Don't we have to handle the case when the user mutes their mic when audio screenshare is in progress ? Don't we have to replace the mixed track with only the desktopAudioStream ?

Reply:
The case where the user mutes his mic during audio screen share is already handled, i.e. the when a track has AudioMixer effect and is muted the mute action is also passed to the effect for it to be handled accordingly.
This is supposed to handle the case where 
1. The user initially joins muted (thus no mic audio track )
2. Shares screen audio ( because there is no mic audio track at this point no mixer is created)
3. Unmutes mic, resulting in a new track being created at line 974
Before this change the screen audio was simply replaced without it being destroyed resulting in an abnormal state and only mic audio being sent. After the change we create a new audio mixer with screen audio and mic audio and use that as the new track.

Reply:
Ow missed that, sorry. I did test it initially with overwrite `config.flags.sendMultipleVideoStreams=true` and from the log line:
`[FeatureFlags] <Object.init>:  Source name signaling: false, Send multiple video streams: true, uses Unified plan: true`
I assume it's activated. But looking in more details it seems that source name signaling needed to be enabled as well for the new flow to go through.
Testing it now I see what you mean, and there might be additional parts of the code that uses `this._desktopAudioStream` I'll do some more edge case testing.

Reply:
Yes for multi-stream mode, both sourceNameSignaling and sendMultipleVideoStreams have to enabled in config.js and in the jvb conf the following line needs to be added.

```
multi-stream {
    enabled = true
 }
```